include/lsst/afw/image/detail/StorableMap.h

```diff
@@ -168,6 +168,38 @@ class StorableMap final {
      */
     bool contains(std::string const& key) const;
 
+    /**
+     * Return `true` if this map contains a mapping for the specified key.
+     *
+     * This is equivalent to testing whether `at(key)` would succeed.
+     *
+     * @tparam T the type of element being tested for
+     * @param key the key to search for
+     *
+     * @return `true` if this map contains a mapping from the specified key to
+     *         a `T`
+     *
+     * @exceptsafe Provides strong exception safety.
+     *
+     */
+    template <typename T, typename std::enable_if_t<std::is_base_of<Storable, T>::value, int> = 0>
+    bool contains(Key<std::shared_ptr<T>> const& key) const {
+        static_assert(std::is_const<T>::value,
+                      "Due to implementation constraints, pointers to non-const are not supported.");
+        if (_contents.count(key.getId()) > 0) {
+            mapped_type const& pointer = _contents.at(key.getId());
+            if (pointer == nullptr) {  // Can't confirm type with dynamic_cast
+                return true;
+            }
+            std::shared_ptr<T> typedPointer = std::dynamic_pointer_cast<T>(pointer);
+            // shared_ptr can be empty without being null. dynamic_pointer_cast
+            // only promises result of failed cast is empty, so test for that
```

Comment:
AFAIK names of components are always left with their native capitalization. Doing otherwise leads to confusion, at least in case-sensitive languages. Python does this too; see the documentation for [`dict.fromkeys`](https://docs.python.org/3/library/stdtypes.html#dict.fromkeys).

Reply:
Fair point on the capitalization. Periods at the end are good though.